wagtail/admin/panels/model_utils.py

```diff
@@ -47,3 +50,31 @@ def get_edit_handler(model):
         panel = ObjectList(panels)
 
     return panel.bind_to_model(model)
+
+
+def expand_panel_list(model, panels):
+    """
+    Given a list which may be a mixture of Panel instances and strings (representing field/relation names),
+    expand it into a flat list of Panel instances
+    """
+    result = []
+    for panel in panels:
+        if isinstance(panel, Panel):
+            result.append(panel)
+
+        elif isinstance(panel, str):
+            field = model._meta.get_field(panel)
+            if isinstance(field, ManyToOneRel):
+                from .inline_panel import InlinePanel
+
+                result.append(InlinePanel(panel))
+            else:
+                result.append(FieldPanel(panel))
+
+        else:
+            raise ImproperlyConfigured(
+                "Invalid panel definition %r - expected Panel or string, got %r"
+                % (panel, type(panel))
+            )
```

Comment:
Another potential gotcha is the classic single-element tuple problem. For example, if you wanted to remove `"other_field"` from the following:

```py
panels = ("my_field", "other_field")
```

you might end up doing

```py
panels = ("my_field")
```

which would raise a rather cryptic

```
django.core.exceptions.FieldDoesNotExist: MyModel has no field named 'm'
```

Would that be something that warrants a helpful error message, e.g. hinting something like "Perhaps you forgot a comma in your definition of `MyModel.panels`?"

Reply:
Sounds like a nice addition, but not sure if it's in scope for this PR - is it any more likely that someone will write `panels = ("my_field")` than writing `panels = (FieldPanel("my_field"))`?

I'd expect it's pretty uncommon for people to use tuples here anyhow, given that the docs have always used lists and people are used to writing `content_panels = Page.content_panels + [...]` (which won't work with a tuple).

Reply:
So I guess the potential improvement here is either:

- Special case the string iterable misconfiguration and raise a more helpful error message in this PR, or
- Catch `FieldDoesNotExist` in our panels construction mechanism and raise a more Wagtail-specific error message, e.g. something like `"Cannot create a panel for field 'field_name' with model 'MyModel', as the field does not exist."`
  - This would be a separate PR, of course, which would be a separate enhancement even without this PR.